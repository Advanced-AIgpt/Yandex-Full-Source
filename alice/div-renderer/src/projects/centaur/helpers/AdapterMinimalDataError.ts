export class AdapterMinimalDataError extends Error {
    constructor(missedData: string[], template: string) {
        super(`Missed data (${missedData.join(', ')}) in template ${template}`);
        this.name = 'AdapterMinimalDataError';
    }
}
