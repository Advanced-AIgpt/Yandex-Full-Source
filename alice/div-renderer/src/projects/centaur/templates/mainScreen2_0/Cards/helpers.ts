export function getCardMainScreenId({ rowIndex, colIndex }: { rowIndex?: number; colIndex?: number }) {
    return `main_screen_card_${rowIndex || 0}_${colIndex || 0}`;
}
