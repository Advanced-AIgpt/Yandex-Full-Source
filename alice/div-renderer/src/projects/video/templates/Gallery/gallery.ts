import {
    TemplateCard,
    Templates,
    ContainerBlock,
    MatchParentSize,
    DivStateBlock,
    SolidBackground,
} from 'divcard2';
import {
    InitCardTemplate,
    GetDocumentTemplate,
    GetIndicatorTemplate,
} from './templates/GalleryTemplates';
import { NAlice } from '../../../../protos';
import { AnimatedText } from './templates/AnimatedText';
import { BaseButton } from './templates/BaseButton';
import { GradientButton } from './templates/GradientButton';
import { StdButton } from './templates/StdButton';
import { TitleImage } from './templates/TitleImage';
import { Title } from './templates/Title';
import { LegalLogo } from './templates/LegalLogo';
import { Description } from './templates/Description';
import { Meta } from './templates/Meta';
import { Subscription } from './templates/Subscription';
import { Rating } from './templates/Rating';
import { Thumbnail } from './templates/Thumbnail';
import { PromoItem } from './templates/PromoItem';
import { ErrorCard } from './templates/ErrorCard';
import { InitCard } from './templates/InitCard';

const templates = {
    animated_text: AnimatedText(),
    base_button: BaseButton(),
    gradient_button: GradientButton(),
    std_button: StdButton({}, '', '', ''),
    title_image: TitleImage(),
    title: Title(),
    legal_logo: LegalLogo(),
    description: Description(),
    meta: Meta(),
    subscription: Subscription(),
    rating: Rating(),
    thumbnail: Thumbnail(),
    promo_item: PromoItem(),
    error_card: ErrorCard(),
    init_card: InitCard(),
};

export const galleryRender = (data: NAlice.NData.ITGalleryData) => {
    if (data.Cards == null) {
        throw new Error('Cards must always be specified');
    }
    return new TemplateCard(new Templates(templates), {
        log_id: 'video.top_div_gallery',
        states: [
            {
                state_id: 1,
                div: new DivStateBlock({
                    height: new MatchParentSize(),
                    width: new MatchParentSize(),
                    div_id: 'screens',
                    states: [
                        {
                            state_id: 'init',
                            div: InitCardTemplate(),
                        },
                        {
                            state_id: 'screen1',
                            div: new ContainerBlock({
                                id: 'ss1',
                                background: [new SolidBackground({
                                    color: '#151517',
                                })],
                                orientation: 'overlap',
                                height: new MatchParentSize(),
                                visibility_action: {
                                    log_id: 'carousel_show',
                                    log_limit: 0,
                                    payload: {
                                        carousel_id: data.GalleryId,
                                        title: data.GalleryTitle,
                                        place: data.GalleryParentScreen,
                                        position: data.GalleryPosition,
                                    },
                                    url: 'metrics://carouselShow',
                                    visibility_duration: 1000,
                                },
                                items: [
                                    new DivStateBlock({
                                        div_id: 'cards',
                                        height: new MatchParentSize(),
                                        states: data.Cards.map((cardData, index) => {
                                            return {
                                                state_id: `card${index}`,
                                                div: GetDocumentTemplate(cardData, data, index),
                                            };
                                        }),
                                    }),
                                    GetIndicatorTemplate(),
                                ],
                            }),
                        },
                    ],
                }),
            },
        ],
    });
};
