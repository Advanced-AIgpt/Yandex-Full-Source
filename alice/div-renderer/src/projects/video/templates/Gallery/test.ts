import { galleryRender } from './gallery';
import { AnonymizeDataForSnapshot } from '../../../../common/helpers/dev';
import { NAlice } from '../../../../protos';

jest.mock('../../../../common/logger');

describe('Video scenario', () => {
    it('matches snapshot', () => {
        const scenarioData: NAlice.NData.ITGalleryData = {
            GalleryId: 'test',
            GalleryTitle: 'Test title',
            GalleryPosition: 0,
            GalleryParentScreen: 'main',
            RequestId: 'reqtst',
            ApphostRequestId: 'apphostreqtst',
            Cards: [
                {
                    ContentId: 'tstcontentid1',
                    ContentType: 'vod-episode',
                    TitleText: 'Тестовый фильм 1',
                    DescriptionText: 'Тестовое описание фильма 1',
                    Rating: 7.5678,
                    LegalLogoAlpha: 0,
                    LogoAlpha: 0,
                    OntoId: 'ruw123',
                    Genres: ['мелодрама', 'детектив'],
                    Countries: 'Канада',
                    ReleaseYear: 2011,
                    LogoUrls: {
                        BaseUrl: 'http://avatars.mds.yandex.net/get-ott/1672343/2a0000017af85caf7cd82a61b509216ee0e2/',
                        Sizes: ['120x90', '400x300', '1920x1080', 'orig'],
                    },
                },
                {
                    ContentId: 'tstcontentid2',
                    ContentType: 'vod-library',
                    TitleText: 'Тестовый фильм 2',
                    DescriptionText: 'Тестовое описание фильма 2',
                    RatingColor: '#123123',
                    Rating: 8.1,
                    LegalLogoAlpha: 0,
                    LogoAlpha: 0,
                    OntoId: 'ruw456',
                    Genres: ['триллер', 'драма'],
                    Countries: 'США',
                    ReleaseYear: 2021,
                    LogoUrls: {
                        BaseUrl: 'http://avatars.mds.yandex.net/get-ott/374297/2a0000017c180dfee9f6c1ee626b004923a1/',
                        Sizes: ['120x90', '400x300', '1920x1080', 'orig'],
                    },
                    LegalLogoUrls: {
                        BaseUrl: 'https://avatars.mds.yandex.net/get-ott/2419418/2a0000017eb9b60b5f8bd4d200217535fe5e/',
                        Sizes: ['orig'],
                    },
                },
            ],
        };
        expect(AnonymizeDataForSnapshot(galleryRender(scenarioData))).toMatchSnapshot();
    });
});
