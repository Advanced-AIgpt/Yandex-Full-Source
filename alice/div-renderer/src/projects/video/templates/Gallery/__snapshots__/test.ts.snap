// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video scenario matches snapshot 1`] = `
Object {
  "card": Object {
    "log_id": "video.top_div_gallery",
    "states": Array [
      Object {
        "div": Object {
          "div_id": "screens",
          "height": Object {
            "type": "match_parent",
          },
          "states": Array [
            Object {
              "div": Object {
                "alpha": 0,
                "height": Object {
                  "type": "match_parent",
                },
                "id": "ss0",
                "text": "Init",
                "text_color": "#170d1f",
                "type": "text",
                "visibility_action": Object {
                  "log_id": "inited",
                  "url": "div-action://set_state?state_id=1/screens/screen1",
                },
                "width": Object {
                  "type": "match_parent",
                },
              },
              "state_id": "init",
            },
            Object {
              "div": Object {
                "background": Array [
                  Object {
                    "color": "#151517",
                    "type": "solid",
                  },
                ],
                "height": Object {
                  "type": "match_parent",
                },
                "id": "ss1",
                "items": Array [
                  Object {
                    "div_id": "cards",
                    "height": Object {
                      "type": "match_parent",
                    },
                    "states": Array [
                      Object {
                        "div": Object {
                          "btns": Array [
                            Object {
                              "action": Object {
                                "log_id": "open_item",
                                "payload": Object {
                                  "content_id": "tstcontentid1",
                                  "content_type": "vod-episode",
                                  "general_analytics_info_payload": Object {},
                                  "on_vh_content_card_opened_event": Object {},
                                  "onto_id": "ruw123",
                                  "title": "Тестовый фильм 1",
                                },
                                "url": "router://openDetails",
                              },
                              "background": Array [
                                Object {
                                  "color": "#333436",
                                  "type": "solid",
                                },
                              ],
                              "border": Object {
                                "corner_radius": 24,
                                "has_shadow": 0,
                              },
                              "extensions": Array [
                                Object {
                                  "id": "request_focus",
                                },
                              ],
                              "focus": Object {
                                "background": Array [
                                  Object {
                                    "color": "#ffffff",
                                    "type": "solid",
                                  },
                                ],
                              },
                              "focused_text_color": "#151517",
                              "font_size": 14,
                              "font_weight": "medium",
                              "margins": Object {
                                "bottom": 5,
                                "left": 5,
                                "right": 5,
                                "top": 5,
                              },
                              "paddings": Object {
                                "bottom": 11,
                                "left": 16,
                                "right": 16,
                                "top": 11,
                              },
                              "text": "Смотреть",
                              "text_alignment_horizontal": "center",
                              "text_color": "#ffffff",
                              "type": "text",
                              "width": Object {
                                "type": "wrap_content",
                              },
                            },
                          ],
                          "description": "Тестовое описание фильма 1",
                          "description_text": "Тестовое описание фильма 1",
                          "has_legal_logo": 0,
                          "has_logo": 1,
                          "id": "item0",
                          "legal_logo_id": "legallogo0",
                          "legal_logo_url": "",
                          "logo_url": "http://avatars.mds.yandex.net/get-ott/1672343/2a0000017af85caf7cd82a61b509216ee0e2/120x90",
                          "meta_id": "meta0",
                          "meta_text": " • мелодрама, детектив • 2011 • Канада",
                          "rating_color": "#89c939",
                          "rating_id": "rating0",
                          "rating_text": "7.6",
                          "subscription_id": "subscription0",
                          "subscription_text": " ",
                          "thumbnail_id": "thumbnail0",
                          "thumbnail_url": "",
                          "title_id": "title0",
                          "title_text": "Тестовый фильм 1",
                          "type": "promo_item",
                          "visibility_action": Object {
                            "log_id": "card_show",
                            "log_limit": 0,
                            "payload": Object {
                              "carousel": Object {
                                "carousel_id": "test",
                                "place": "main",
                                "position": 0,
                                "title": "Test title",
                              },
                              "content_item": Object {
                                "content_id": "tstcontentid1",
                                "content_type": "vod-episode",
                                "req_info": Object {
                                  "apphost-reqid": "apphostreqtst",
                                  "reqid": "reqtst",
                                },
                              },
                              "position": 0,
                            },
                            "url": "metrics://cardShow",
                            "visibility_duration": 1000,
                          },
                        },
                        "state_id": "card0",
                      },
                      Object {
                        "div": Object {
                          "btns": Array [
                            Object {
                              "action": Object {
                                "log_id": "open_item",
                                "payload": Object {
                                  "content_id": "tstcontentid2",
                                  "content_type": "vod-library",
                                  "general_analytics_info_payload": Object {},
                                  "on_vh_content_card_opened_event": Object {},
                                  "onto_id": "ruw456",
                                  "title": "Тестовый фильм 2",
                                },
                                "url": "router://openDetails",
                              },
                              "background": Array [
                                Object {
                                  "color": "#333436",
                                  "type": "solid",
                                },
                              ],
                              "border": Object {
                                "corner_radius": 24,
                                "has_shadow": 0,
                              },
                              "extensions": Array [
                                Object {
                                  "id": "request_focus",
                                },
                              ],
                              "focus": Object {
                                "background": Array [
                                  Object {
                                    "color": "#ffffff",
                                    "type": "solid",
                                  },
                                ],
                              },
                              "focused_text_color": "#151517",
                              "font_size": 14,
                              "font_weight": "medium",
                              "margins": Object {
                                "bottom": 5,
                                "left": 5,
                                "right": 5,
                                "top": 5,
                              },
                              "paddings": Object {
                                "bottom": 11,
                                "left": 16,
                                "right": 16,
                                "top": 11,
                              },
                              "text": "Смотреть",
                              "text_alignment_horizontal": "center",
                              "text_color": "#ffffff",
                              "type": "text",
                              "width": Object {
                                "type": "wrap_content",
                              },
                            },
                          ],
                          "description": "Тестовое описание фильма 2",
                          "description_text": "Тестовое описание фильма 2",
                          "has_legal_logo": 1,
                          "has_logo": 1,
                          "id": "item1",
                          "legal_logo_id": "legallogo1",
                          "legal_logo_url": "https://avatars.mds.yandex.net/get-ott/2419418/2a0000017eb9b60b5f8bd4d200217535fe5e/orig",
                          "logo_url": "http://avatars.mds.yandex.net/get-ott/374297/2a0000017c180dfee9f6c1ee626b004923a1/120x90",
                          "meta_id": "meta1",
                          "meta_text": " • триллер, драма • 2021 • США",
                          "rating_color": "#32ba43",
                          "rating_id": "rating1",
                          "rating_text": "8.1",
                          "subscription_id": "subscription1",
                          "subscription_text": " ",
                          "thumbnail_id": "thumbnail1",
                          "thumbnail_url": "",
                          "title_id": "title1",
                          "title_text": "Тестовый фильм 2",
                          "type": "promo_item",
                          "visibility_action": Object {
                            "log_id": "card_show",
                            "log_limit": 0,
                            "payload": Object {
                              "carousel": Object {
                                "carousel_id": "test",
                                "place": "main",
                                "position": 0,
                                "title": "Test title",
                              },
                              "content_item": Object {
                                "content_id": "tstcontentid2",
                                "content_type": "vod-library",
                                "req_info": Object {
                                  "apphost-reqid": "apphostreqtst",
                                  "reqid": "reqtst",
                                },
                              },
                              "position": 1,
                            },
                            "url": "metrics://cardShow",
                            "visibility_duration": 1000,
                          },
                        },
                        "state_id": "card1",
                      },
                    ],
                    "type": "state",
                  },
                  Object {
                    "alignment_horizontal": "right",
                    "alignment_vertical": "bottom",
                    "custom_props": Object {
                      "color": "#333333",
                      "count": 10,
                      "selected_color": "#ffffff",
                      "size": 9,
                      "space": 4,
                      "visible_count": 4,
                    },
                    "custom_type": "indicator",
                    "margins": Object {
                      "bottom": 244,
                      "right": 36,
                    },
                    "type": "custom",
                    "width": Object {
                      "type": "wrap_content",
                    },
                  },
                ],
                "orientation": "overlap",
                "type": "container",
                "visibility_action": Object {
                  "log_id": "carousel_show",
                  "log_limit": 0,
                  "payload": Object {
                    "carousel_id": "test",
                    "place": "main",
                    "position": 0,
                    "title": "Test title",
                  },
                  "url": "metrics://carouselShow",
                  "visibility_duration": 1000,
                },
              },
              "state_id": "screen1",
            },
          ],
          "type": "state",
          "width": Object {
            "type": "match_parent",
          },
        },
        "state_id": 1,
      },
    ],
  },
  "templates": Object {
    "animated_text": Object {
      "transition_in": Object {
        "alpha": 0,
        "duration": 300,
        "interpolator": "ease_out",
        "start_delay": 150,
        "type": "fade",
      },
      "transition_out": Object {
        "alpha": 0,
        "duration": 150,
        "interpolator": "ease_in",
        "type": "fade",
      },
      "type": "text",
    },
    "base_button": Object {
      "$text": "button_text",
      "action": Object {
        "$log_id": "button_log_id",
        "$payload": "action_payload",
        "$url": "button_action",
      },
      "background": Array [
        Object {
          "color": "#333436",
          "type": "solid",
        },
      ],
      "border": Object {
        "corner_radius": 24,
        "has_shadow": 0,
      },
      "extensions": Array [
        Object {
          "id": "request_focus",
        },
      ],
      "font_size": 14,
      "font_weight": "medium",
      "margins": Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      },
      "paddings": Object {
        "bottom": 11,
        "left": 16,
        "right": 16,
        "top": 11,
      },
      "text_alignment_horizontal": "center",
      "text_color": "#ffffff",
      "type": "text",
      "width": Object {
        "type": "wrap_content",
      },
    },
    "description": Object {
      "font_size": 16,
      "font_weight": "regular",
      "height": Object {
        "type": "fixed",
        "value": 70,
      },
      "line_height": 20,
      "margins": Object {
        "left": 20,
        "right": 20,
        "top": 3,
      },
      "max_lines": 3,
      "text_color": "#ffffff",
      "transition_in": Object {
        "alpha": 0,
        "duration": 300,
        "interpolator": "ease_out",
        "start_delay": 150,
        "type": "fade",
      },
      "transition_out": Object {
        "alpha": 0,
        "duration": 150,
        "interpolator": "ease_in",
        "type": "fade",
      },
      "truncate": "end",
      "type": "text",
      "width": Object {
        "type": "fixed",
        "value": 400,
      },
    },
    "error_card": Object {
      "height": Object {
        "type": "match_parent",
      },
      "items": Array [
        Object {
          "$id": "thumbnail_id",
          "$image_url": "thumbnail_url",
          "alignment_horizontal": "right",
          "alignment_vertical": "top",
          "extensions": Array [
            Object {
              "id": "transition",
              "params": Object {
                "name": "thumbnail",
                "target": "thumbnail",
              },
            },
          ],
          "height": Object {
            "type": "fixed",
            "value": 400,
          },
          "transition_in": Object {
            "alpha": 0,
            "duration": 400,
            "interpolator": "ease_out",
            "type": "fade",
          },
          "type": "image",
          "width": Object {
            "type": "fixed",
            "value": 700,
          },
        },
        Object {
          "background": Array [
            Object {
              "angle": 0,
              "colors": Array [
                "#ff151517",
                "#ff151517",
                "#ff151517",
                "#c0151517",
                "#80151517",
                "#30151517",
                "#00151517",
                "#00151517",
                "#00151517",
              ],
              "type": "gradient",
            },
            Object {
              "angle": 90,
              "colors": Array [
                "#ff151517",
                "#ff151517",
                "#ff151517",
                "#b0151517",
                "#70151517",
                "#10151517",
                "#00151517",
                "#00151517",
              ],
              "type": "gradient",
            },
          ],
          "content_alignment_horizontal": "left",
          "content_alignment_vertical": "bottom",
          "height": Object {
            "type": "match_parent",
          },
          "items": Array [
            Object {
              "$text": "title_text",
              "alignment_horizontal": "left",
              "font_size": 36,
              "font_weight": "bold",
              "height": Object {
                "type": "match_parent",
              },
              "line_height": 40,
              "margins": Object {
                "left": 20,
                "top": 23,
              },
              "max_lines": 2,
              "text_alignment_horizontal": "left",
              "text_alignment_vertical": "bottom",
              "text_color": "#ffffff",
              "transition_in": Object {
                "alpha": 0,
                "duration": 300,
                "interpolator": "ease_out",
                "start_delay": 150,
                "type": "fade",
              },
              "transition_out": Object {
                "alpha": 0,
                "duration": 150,
                "interpolator": "ease_in",
                "type": "fade",
              },
              "truncate": "end",
              "type": "text",
            },
            Object {
              "$id": "description_id",
              "$text": "description_text",
              "font_size": 16,
              "font_weight": "regular",
              "height": Object {
                "type": "fixed",
                "value": 70,
              },
              "line_height": 20,
              "margins": Object {
                "left": 20,
                "right": 20,
                "top": 12,
              },
              "max_lines": 3,
              "text_color": "#ffffff",
              "transition_in": Object {
                "alpha": 0,
                "duration": 300,
                "interpolator": "ease_out",
                "start_delay": 150,
                "type": "fade",
              },
              "transition_out": Object {
                "alpha": 0,
                "duration": 150,
                "interpolator": "ease_in",
                "type": "fade",
              },
              "truncate": "end",
              "type": "text",
              "width": Object {
                "type": "fixed",
                "value": 400,
              },
            },
            Object {
              "$items": "btns",
              "margins": Object {
                "bottom": 232,
                "left": 15,
                "right": 15,
                "top": 66,
              },
              "orientation": "horizontal",
              "type": "container",
            },
          ],
          "type": "container",
        },
      ],
      "orientation": "overlap",
      "type": "container",
    },
    "gradient_button": Object {
      "$text": "button_text",
      "action": Object {
        "$log_id": "button_log_id",
        "$payload": "action_payload",
        "$url": "button_action",
      },
      "background": Array [
        Object {
          "color": "#333436",
          "type": "solid",
        },
      ],
      "border": Object {
        "corner_radius": 24,
        "has_shadow": 0,
      },
      "extensions": Array [
        Object {
          "id": "request_focus",
        },
      ],
      "focus": Object {
        "background": Array [
          Object {
            "angle": 190,
            "colors": Array [
              "#F19743",
              "#AF44C4",
              "#5259DC",
              "#505ADD",
            ],
            "type": "gradient",
          },
        ],
      },
      "focused_text_color": "#ffffff",
      "font_size": 14,
      "font_weight": "medium",
      "margins": Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      },
      "paddings": Object {
        "bottom": 11,
        "left": 16,
        "right": 16,
        "top": 11,
      },
      "text_alignment_horizontal": "center",
      "text_color": "#ffffff",
      "type": "text",
      "width": Object {
        "type": "wrap_content",
      },
    },
    "init_card": Object {
      "alpha": 0,
      "height": Object {
        "type": "match_parent",
      },
      "id": "ss0",
      "text": "Init",
      "text_color": "#170d1f",
      "type": "text",
      "visibility_action": Object {
        "log_id": "inited",
        "url": "div-action://set_state?state_id=1/screens/screen1",
      },
      "width": Object {
        "type": "match_parent",
      },
    },
    "legal_logo": Object {
      "alignment_horizontal": "right",
      "alignment_vertical": "top",
      "content_alignment_horizontal": "right",
      "content_alignment_vertical": "top",
      "height": Object {
        "type": "fixed",
        "value": 60,
      },
      "margins": Object {
        "right": 30,
        "top": 52,
      },
      "scale": "fit",
      "type": "image",
      "width": Object {
        "type": "wrap_content",
      },
    },
    "meta": Object {
      "font_size": 14,
      "font_weight": "medium",
      "max_lines": 1,
      "text_color": "#b3b3b3",
      "transition_in": Object {
        "alpha": 0,
        "duration": 300,
        "interpolator": "ease_out",
        "start_delay": 150,
        "type": "fade",
      },
      "transition_out": Object {
        "alpha": 0,
        "duration": 150,
        "interpolator": "ease_in",
        "type": "fade",
      },
      "truncate": "end",
      "type": "text",
      "width": Object {
        "type": "fixed",
        "value": 400,
      },
    },
    "promo_item": Object {
      "height": Object {
        "type": "match_parent",
      },
      "items": Array [
        Object {
          "$id": "thumbnail_id",
          "$image_url": "thumbnail_url",
          "alignment_horizontal": "right",
          "alignment_vertical": "top",
          "extensions": Array [
            Object {
              "id": "transition",
              "params": Object {
                "name": "thumbnail",
                "target": "thumbnail",
              },
            },
          ],
          "height": Object {
            "type": "fixed",
            "value": 400,
          },
          "transition_in": Object {
            "alpha": 0,
            "duration": 400,
            "interpolator": "ease_out",
            "type": "fade",
          },
          "type": "image",
          "width": Object {
            "type": "fixed",
            "value": 700,
          },
        },
        Object {
          "background": Array [
            Object {
              "angle": 0,
              "colors": Array [
                "#ff151517",
                "#ff151517",
                "#ff151517",
                "#c0151517",
                "#80151517",
                "#30151517",
                "#02151517",
                "#02151517",
                "#03151517",
              ],
              "type": "gradient",
            },
            Object {
              "angle": 90,
              "colors": Array [
                "#ff151517",
                "#ff151517",
                "#ff151517",
                "#b0151517",
                "#70151517",
                "#10151517",
                "#02151517",
                "#02151517",
              ],
              "type": "gradient",
            },
          ],
          "content_alignment_horizontal": "left",
          "content_alignment_vertical": "bottom",
          "height": Object {
            "type": "match_parent",
          },
          "items": Array [
            Object {
              "content_alignment_horizontal": "left",
              "content_alignment_vertical": "bottom",
              "height": Object {
                "type": "match_parent",
              },
              "items": Array [
                Object {
                  "$id": "title_id",
                  "$text": "title_text",
                  "alignment_horizontal": "left",
                  "font_size": 36,
                  "font_weight": "bold",
                  "height": Object {
                    "type": "match_parent",
                  },
                  "line_height": 40,
                  "margins": Object {
                    "left": 20,
                    "top": 23,
                  },
                  "max_lines": 2,
                  "text_alignment_horizontal": "left",
                  "text_alignment_vertical": "bottom",
                  "text_color": "#ffffff",
                  "transition_in": Object {
                    "alpha": 0,
                    "duration": 300,
                    "interpolator": "ease_out",
                    "start_delay": 150,
                    "type": "fade",
                  },
                  "transition_out": Object {
                    "alpha": 0,
                    "duration": 150,
                    "interpolator": "ease_in",
                    "type": "fade",
                  },
                  "truncate": "end",
                  "type": "text",
                },
                Object {
                  "$alpha": "has_logo",
                  "$image_url": "logo_url",
                  "alignment_horizontal": "left",
                  "alignment_vertical": "bottom",
                  "content_alignment_horizontal": "left",
                  "content_alignment_vertical": "bottom",
                  "extensions": Array [
                    Object {
                      "id": "transition",
                      "params": Object {
                        "name": "logo",
                        "target": "logo",
                      },
                    },
                  ],
                  "height": Object {
                    "type": "match_parent",
                  },
                  "id": "logo_id",
                  "margins": Object {
                    "left": 20,
                    "right": 20,
                    "top": 23,
                  },
                  "scale": "fit",
                  "transition_in": Object {
                    "alpha": 0,
                    "duration": 300,
                    "interpolator": "ease_out",
                    "start_delay": 150,
                    "type": "fade",
                  },
                  "transition_out": Object {
                    "alpha": 0,
                    "duration": 150,
                    "interpolator": "ease_in",
                    "type": "fade",
                  },
                  "type": "image",
                },
              ],
              "orientation": "overlap",
              "type": "container",
              "width": Object {
                "type": "fixed",
                "value": 400,
              },
            },
            Object {
              "items": Array [
                Object {
                  "$id": "rating_id",
                  "$text": "rating_text",
                  "$text_color": "rating_color",
                  "font_size": 14,
                  "font_weight": "medium",
                  "max_lines": 1,
                  "transition_in": Object {
                    "alpha": 0,
                    "duration": 300,
                    "interpolator": "ease_out",
                    "start_delay": 150,
                    "type": "fade",
                  },
                  "transition_out": Object {
                    "alpha": 0,
                    "duration": 150,
                    "interpolator": "ease_in",
                    "type": "fade",
                  },
                  "type": "text",
                },
                Object {
                  "$id": "meta_id",
                  "$text": "meta_text",
                  "font_size": 14,
                  "font_weight": "medium",
                  "max_lines": 1,
                  "text_color": "#b3b3b3",
                  "transition_in": Object {
                    "alpha": 0,
                    "duration": 300,
                    "interpolator": "ease_out",
                    "start_delay": 150,
                    "type": "fade",
                  },
                  "transition_out": Object {
                    "alpha": 0,
                    "duration": 150,
                    "interpolator": "ease_in",
                    "type": "fade",
                  },
                  "truncate": "end",
                  "type": "text",
                  "width": Object {
                    "type": "fixed",
                    "value": 400,
                  },
                },
              ],
              "margins": Object {
                "left": 20,
                "right": 20,
                "top": 12,
              },
              "orientation": "horizontal",
              "type": "container",
              "width": Object {
                "type": "wrap_content",
              },
            },
            Object {
              "$id": "description",
              "$text": "description_text",
              "font_size": 16,
              "font_weight": "regular",
              "height": Object {
                "type": "fixed",
                "value": 70,
              },
              "line_height": 20,
              "margins": Object {
                "left": 20,
                "right": 20,
                "top": 3,
              },
              "max_lines": 3,
              "text_color": "#ffffff",
              "transition_in": Object {
                "alpha": 0,
                "duration": 300,
                "interpolator": "ease_out",
                "start_delay": 150,
                "type": "fade",
              },
              "transition_out": Object {
                "alpha": 0,
                "duration": 150,
                "interpolator": "ease_in",
                "type": "fade",
              },
              "truncate": "end",
              "type": "text",
              "width": Object {
                "type": "fixed",
                "value": 400,
              },
            },
            Object {
              "$id": "subscription_id",
              "$text": "subscription_text",
              "font_size": 14,
              "font_weight": "regular",
              "margins": Object {
                "left": 20,
                "right": 20,
                "top": 18,
              },
              "max_lines": 1,
              "text_color": "#b3b3b3",
              "transition_in": Object {
                "alpha": 0,
                "duration": 300,
                "interpolator": "ease_out",
                "start_delay": 150,
                "type": "fade",
              },
              "transition_out": Object {
                "alpha": 0,
                "duration": 150,
                "interpolator": "ease_in",
                "type": "fade",
              },
              "truncate": "end",
              "type": "text",
              "width": Object {
                "type": "fixed",
                "value": 400,
              },
            },
            Object {
              "$items": "btns",
              "margins": Object {
                "bottom": 232,
                "left": 15,
                "right": 15,
                "top": 7,
              },
              "orientation": "horizontal",
              "type": "container",
            },
          ],
          "type": "container",
        },
        Object {
          "$alpha": "has_legal_logo",
          "$id": "legal_logo_id",
          "$image_url": "legal_logo_url",
          "alignment_horizontal": "right",
          "alignment_vertical": "top",
          "content_alignment_horizontal": "right",
          "content_alignment_vertical": "top",
          "height": Object {
            "type": "fixed",
            "value": 60,
          },
          "margins": Object {
            "right": 30,
            "top": 52,
          },
          "scale": "fit",
          "type": "image",
          "width": Object {
            "type": "wrap_content",
          },
        },
      ],
      "orientation": "overlap",
      "type": "container",
    },
    "rating": Object {
      "font_size": 14,
      "font_weight": "medium",
      "max_lines": 1,
      "transition_in": Object {
        "alpha": 0,
        "duration": 300,
        "interpolator": "ease_out",
        "start_delay": 150,
        "type": "fade",
      },
      "transition_out": Object {
        "alpha": 0,
        "duration": 150,
        "interpolator": "ease_in",
        "type": "fade",
      },
      "type": "text",
    },
    "std_button": Object {
      "action": Object {
        "log_id": "",
        "payload": Object {
          "content_id": "",
          "content_type": "",
          "general_analytics_info_payload": Object {},
          "on_vh_content_card_opened_event": Object {},
          "onto_id": "",
          "title": "",
        },
        "url": "",
      },
      "background": Array [
        Object {
          "color": "#333436",
          "type": "solid",
        },
      ],
      "border": Object {
        "corner_radius": 24,
        "has_shadow": 0,
      },
      "extensions": Array [
        Object {
          "id": "request_focus",
        },
      ],
      "focus": Object {
        "background": Array [
          Object {
            "color": "#ffffff",
            "type": "solid",
          },
        ],
      },
      "focused_text_color": "#151517",
      "font_size": 14,
      "font_weight": "medium",
      "margins": Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      },
      "paddings": Object {
        "bottom": 11,
        "left": 16,
        "right": 16,
        "top": 11,
      },
      "text": "",
      "text_alignment_horizontal": "center",
      "text_color": "#ffffff",
      "type": "text",
      "width": Object {
        "type": "wrap_content",
      },
    },
    "subscription": Object {
      "font_size": 14,
      "font_weight": "regular",
      "margins": Object {
        "left": 20,
        "right": 20,
        "top": 18,
      },
      "max_lines": 1,
      "text_color": "#b3b3b3",
      "transition_in": Object {
        "alpha": 0,
        "duration": 300,
        "interpolator": "ease_out",
        "start_delay": 150,
        "type": "fade",
      },
      "transition_out": Object {
        "alpha": 0,
        "duration": 150,
        "interpolator": "ease_in",
        "type": "fade",
      },
      "truncate": "end",
      "type": "text",
      "width": Object {
        "type": "fixed",
        "value": 400,
      },
    },
    "thumbnail": Object {
      "alignment_horizontal": "right",
      "alignment_vertical": "top",
      "extensions": Array [
        Object {
          "id": "transition",
          "params": Object {
            "name": "thumbnail",
            "target": "thumbnail",
          },
        },
      ],
      "height": Object {
        "type": "fixed",
        "value": 400,
      },
      "transition_in": Object {
        "alpha": 0,
        "duration": 400,
        "interpolator": "ease_out",
        "type": "fade",
      },
      "type": "image",
      "width": Object {
        "type": "fixed",
        "value": 700,
      },
    },
    "title": Object {
      "alignment_horizontal": "left",
      "font_size": 36,
      "font_weight": "bold",
      "height": Object {
        "type": "match_parent",
      },
      "line_height": 40,
      "margins": Object {
        "left": 20,
        "top": 23,
      },
      "max_lines": 2,
      "text_alignment_horizontal": "left",
      "text_alignment_vertical": "bottom",
      "text_color": "#ffffff",
      "transition_in": Object {
        "alpha": 0,
        "duration": 300,
        "interpolator": "ease_out",
        "start_delay": 150,
        "type": "fade",
      },
      "transition_out": Object {
        "alpha": 0,
        "duration": 150,
        "interpolator": "ease_in",
        "type": "fade",
      },
      "truncate": "end",
      "type": "text",
    },
    "title_image": Object {
      "alignment_horizontal": "left",
      "alignment_vertical": "bottom",
      "content_alignment_horizontal": "left",
      "content_alignment_vertical": "bottom",
      "extensions": Array [
        Object {
          "id": "transition",
          "params": Object {
            "name": "logo",
            "target": "logo",
          },
        },
      ],
      "height": Object {
        "type": "match_parent",
      },
      "margins": Object {
        "left": 20,
        "right": 20,
        "top": 23,
      },
      "scale": "fit",
      "transition_in": Object {
        "alpha": 0,
        "duration": 300,
        "interpolator": "ease_out",
        "start_delay": 150,
        "type": "fade",
      },
      "transition_out": Object {
        "alpha": 0,
        "duration": 150,
        "interpolator": "ease_in",
        "type": "fade",
      },
      "type": "image",
    },
  },
}
`;
